### ISSUE TITLE
Codeblock highlight incorrectly for rust 'lifetime specifier'.

### ISSUE THREAD

**Issue by sshelll** [hexojs/hexo#5595]
### Check List

- [X] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting).
- [X] I have already searched existing issues and they are not help to me.
- [X] I examined error or warning messages and it's difficult to solve.
- [X] I am using the [latest](https://github.com/hexojs/hexo/releases) version of Hexo. (run `hexo version` to check)
- [X] My Node.js is matched [the required version](https://hexo.io/docs/#Required-Node-js-version).

### Expected behavior

Highlight rust code correctly.

I've tested on the official demo site of highlightjs, it's correct:

![image](https://github.com/user-attachments/assets/e41ef41d-260e-40ea-ad92-48495cd1d796)

### Actual behavior

For this snippet:
```rust
fn main() {
    let hello: &'static str = "hello";
    {
        let world = String::from("world");
        let world = &world;
        debug(hello, world);
    }
}
```
Since the lifetime specifier for rust starts with an 「'」, hexo / highlightjs seems to treat all the characters after it to a single string:
![image](https://github.com/user-attachments/assets/17c8e8bb-6e1d-498a-880f-bfc4630e11f7)

### How to reproduce?

1. hexo init
2. hexo new post test
3. put the rust code snippet in to a codeblock
4. hexo generate && hexo server

### Is the problem still there under `Safe mode`?

Yes!

### Your Node.js & npm version

```text
node -v && npm -v
v23.5.0
10.9.2
```

### Your Hexo and Plugin version

```text
npm ls --depth 0
hexo-site@0.0.0 /Users/xxxx/tmp/hexo_test/test
├── hexo-generator-archive@2.0.0
├── hexo-generator-category@2.0.0
├── hexo-generator-index@4.0.0
├── hexo-generator-tag@2.0.0
├── hexo-renderer-ejs@2.0.0
├── hexo-renderer-marked@6.3.0
├── hexo-renderer-stylus@3.0.1
├── hexo-server@3.0.0
├── hexo-theme-landscape@1.0.0
└── hexo@7.3.0
```

### Your `package.json`

```json
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "hexo generate",
    "clean": "hexo clean",
    "deploy": "hexo deploy",
    "server": "hexo server"
  },
  "hexo": {
    "version": "7.3.0"
  },
  "dependencies": {
    "hexo": "^7.3.0",
    "hexo-generator-archive": "^2.0.0",
    "hexo-generator-category": "^2.0.0",
    "hexo-generator-index": "^4.0.0",
    "hexo-generator-tag": "^2.0.0",
    "hexo-renderer-ejs": "^2.0.0",
    "hexo-renderer-marked": "^6.3.0",
    "hexo-renderer-stylus": "^3.0.1",
    "hexo-server": "^3.0.0",
    "hexo-theme-landscape": "^1.0.0"
  }
}
```

### Your site's `_config.yml` (Optional)

```yaml
default, unchanged
```

**Comment by D-Sketon** [comment]
https://github.com/highlightjs/highlight.js/issues/4190

**Comment by sshelll** [comment]
> https://github.com/highlightjs/highlight.js/issues/4190

Thanks，I'll closed the issue and track that one.

BTW, I wonder if any changes should be made in hexo conf after highlightjs fixed that issue?

**Comment by sshelll** [comment]
highlightjs fixed that issue in v11.11.1

run `npm update` in hexo dir works for me.
